rpc/src/client/transport/http.rs

```diff
@@ -39,6 +48,14 @@ use crate::{client::Client, prelude::*, Error, Scheme, SimpleRequest, Url};
 #[derive(Debug, Clone)]
 pub struct HttpClient {
     inner: sealed::HttpClient,
+    compat: CompatMode,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct HttpConfig {
+    /// Compatibility mode for Tendermint RPC protocol.
+    pub compat: CompatMode,
+    pub proxy_url: Option<HttpClientUrl>,
 }
```

Comment:
This is not the best design for maintainability, and perhaps a better solution would be a builder pattern for HttpClient.

Reply:
Redone in https://github.com/informalsystems/tendermint-rs/pull/1193/commits/84d9cad8c8e41e9e0343633a06ae8288101b0ad3